File: backend/app/schemas/intelligence.py
from typing import Literal

from pydantic import BaseModel

ValidityStatus = Literal["valid", "expiring", "expired", "unknown"]
RequirementStatus = Literal["satisfied", "missing", "expired", "unknown"]
ReadinessStatus = Literal["ready", "almost_ready", "needs_confirmation", "not_ready"]
UsableResult = Literal["likely_acceptable", "needs_confirmation", "expired", "does_not_satisfy"]


class ValidityRead(BaseModel):
    issueDate: str | None = None
    expiryDate: str | None = None
    status: ValidityStatus
    daysLeft: int | None = None
    source: str
    confidence: float


class ConsistencyIssue(BaseModel):
    type: str
    severity: Literal["info", "warning", "high"]
    documents: list[str]
    documentTitles: list[str]
    values: list[str]
    field: str


class ConsistencyRead(BaseModel):
    issues: list[ConsistencyIssue] = []


class RequirementReadiness(BaseModel):
    index: int
    label: str
    status: RequirementStatus
    matchedDocumentId: str | None = None
    matchedDocType: str | None = None
    reason: str
    confidence: float


class ReadinessRead(BaseModel):
    serviceId: str
    status: ReadinessStatus
    score: int
    satisfied: int
    total: int
    requirements: list[RequirementReadiness]


class UsableForRead(BaseModel):
    serviceId: str
    requirement: str | None = None
    result: UsableResult
    reason: str


class DiscoveryService(BaseModel):
    serviceId: str
    status: Literal["ready", "likely_relevant"]
    score: int
    satisfied: int
    total: int
    missingCount: int


class DiscoveryRead(BaseModel):
    services: list[DiscoveryService] = []

File: frontend/types/intelligence.ts
export type ValidityStatus = 'valid' | 'expiring' | 'expired' | 'unknown';

export interface DocumentValidity {
  issueDate: string | null;
  expiryDate: string | null;
  status: ValidityStatus;
  daysLeft: number | null;
  source: string;
  confidence: number;
}

export interface ConsistencyIssue {
  type: string;
  severity: 'info' | 'warning' | 'high';
  documents: string[];
  documentTitles: string[];
  values: string[];
  field: string;
}

export interface ConsistencyResult {
  issues: ConsistencyIssue[];
}

export type RequirementStatus = 'satisfied' | 'missing' | 'expired' | 'unknown';
export type ReadinessStatus = 'ready' | 'almost_ready' | 'needs_confirmation' | 'not_ready';

export interface RequirementReadiness {
  index: number;
  label: string;
  status: RequirementStatus;
  matchedDocumentId: string | null;
  matchedDocType: string | null;
  reason: string;
  confidence: number;
}

export interface Readiness {
  serviceId: string;
  status: ReadinessStatus;
  score: number;
  satisfied: number;
  total: number;
  requirements: RequirementReadiness[];
}

export interface DiscoveryService {
  serviceId: string;
  status: 'ready' | 'likely_relevant';
  score: number;
  satisfied: number;
  total: number;
  missingCount: number;
}

export interface DiscoveryResult {
  services: DiscoveryService[];
}
